ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/correction/RemoveBuildEntryResolution.java

```diff
@@ -57,4 +59,27 @@ protected void createChange(Build build) {
 		}
 	}
 
+	@Override
+	public void run(IMarker marker) {
+		super.run(marker);
+		this.marker = marker;
+	}
+
+	@Override
+	public IMarker[] findOtherMarkers(IMarker[] markers) {
+		HashSet<IMarker> mset = new HashSet<>(markers.length);
+		for (IMarker iMarker : markers) {
+			if (iMarker.equals(this.marker))
+				continue;
+			String str = iMarker.getAttribute(PDEMarkerFactory.compilerKey, ""); //$NON-NLS-1$
+			// RemoveBuildEntryResolution can happen for these 4 compiler flags
+			if (str.equals(CompilerFlags.P_BUILD_SOURCE_LIBRARY) || str.equals(CompilerFlags.P_BUILD_SRC_INCLUDES)
+					|| str.equals(CompilerFlags.P_BUILD_BIN_INCLUDES)
+					|| str.equals(CompilerFlags.P_BUILD_OUTPUT_LIBRARY))
+				mset.add(iMarker);
+		}
+		int size = mset.size();
+		return mset.toArray(new IMarker[size]);
+	}
```

Comment:
When you store the relevant compiler flags in a static Set, it could become even simple:
```suggestion
  private static final Set<String> RELEVANT_COMPILER_FLAGS = Set.of(CompilerFlags.P_BUILD_SOURCE_LIBRARY,
      CompilerFlags.P_BUILD_SRC_INCLUDES, CompilerFlags.P_BUILD_BIN_INCLUDES, CompilerFlags.P_BUILD_OUTPUT_LIBRARY);

  @Override
  public IMarker[] findOtherMarkers(IMarker[] markers) {
    return Arrays.stream(markers).filter(m -> !m.equals(this.marker))
        .filter(m -> RELEVANT_COMPILER_FLAGS.contains(m.getAttribute(PDEMarkerFactory.compilerKey, "")))
        .toArray(IMarker[]::new);
  }
```